File: youkai-front/src/domain/youkai/checkdigit/state/CodeToCharacterConverter.ts
/**
 * 文字コード変換テーブル。
 */
export default class CodeToCharacterConverter {
    private static readonly A_TO_Y = "AHOV16**BIPW27**CJQX38**DKRY49**ELSZ50**FMT-n!**GNU.mc".split("");
    private static readonly INVALID_CHAR = "*";

    public convert(code: number): string {
        return CodeToCharacterConverter.A_TO_Y[code];
    }

    public isInvalidCharCode(code: number): boolean {
        return CodeToCharacterConverter.INVALID_CHAR === this.convert(code);
    }
}

File: src/main/java/com/github/kazuhitom/youkaicheckdigittry/state/CodeToCharacterConverter.java
package com.github.kazuhitom.youkaicheckdigittry.state;

/**
 * 文字コード変換テーブル。
 */
public class CodeToCharacterConverter {
    private static final char[] A_TO_Y = "AHOV16**BIPW27**CJQX38**DKRY49**ELSZ50**FMT-n!**GNU.mc".toCharArray();

    private static final char INVALID_CHAR = '*';

    public char convert(int code) {
        return A_TO_Y[code];
    }

    public boolean isInvalidCharCode(int code) {
        return INVALID_CHAR == convert(code);
    }
}
